### PR TITLE
GEODE-7717: ClusterManagementListResult should show a list of EntityInfo

### PR DESCRIPTION
**Pull request by jinmeiliao**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jinmeiliao** [comment]
@jmelchio

**Comment by jinmeiliao** [comment]
> Pushed small change to remove warnings. Otherwise LGTM.

Did you make changes to my branch? Sorry I force pushed and may have overwritten yours.

**Comment by jmelchio** [comment]
> > Pushed small change to remove warnings. Otherwise LGTM.
> 
> Did you make changes to my branch? Sorry I force pushed and may have overwritten yours.

Yes, made the changes to your branch.

File: geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java [hunk 1]
```diff
@@ -39,13 +40,13 @@
   private String id;
   @JsonInclude
   @JsonProperty
-  private List<EntityGroupInfo<T, R>> configurationByGroup = new ArrayList<>();
+  private List<EntityGroupInfo<T, R>> groups = new ArrayList<>();
 
   public EntityInfo() {}
 
-  public EntityInfo(String id, List<EntityGroupInfo<T, R>> configurationByGroup) {
+  public EntityInfo(String id, List<EntityGroupInfo<T, R>> groups) {
```
Comment (demery-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java]: Would `groupInfo` be a better name for this field (and the corresponding getter and setter)?

File: geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java [hunk 2]
```diff
@@ -88,19 +97,19 @@ public boolean equals(Object o) {
...
-    return Objects.hash(id, configurationByGroup);
+    return Objects.hash(id, groups);
   }
 
   @Override
   public String toString() {
     return "ConfigurationInfo{" +
         "id='" + id + '\'' +
-        ", configurationByGroup=" + configurationByGroup +
+        ", configurationByGroup=" + groups +
```
Comment (demery-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java]: Whatever you decide to name the field, change the label here to match.

File: geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java [hunk 3]
```diff
@@ -56,28 +57,36 @@ public void setId(String id) {
...
   public List<R> getRuntimeInfos() {
-    return configurationByGroup.stream()
+    return groups.stream()
         .flatMap(r -> r.getRuntimeInfo().stream())
         .collect(toList());
   }
 
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+  public Links getLinks() {
+    if (groups.size() == 0) {
```
Comment (dschneider-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java]: groups.isEmpty() is a little better

File: geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementListResult.java [hunk 4]
```diff
@@ -34,50 +34,72 @@
...
-   */
   public ClusterManagementListResult(StatusCode statusCode, String message) {
     super(statusCode, message);
   }
 
-  // Override the mapper setting so that we always show result
-  @JsonInclude
-  @JsonProperty
-  private List<EntityGroupInfo<T, R>> result = new ArrayList<>();
+  private Map<String, EntityInfo<T, R>> entities = new HashMap<>();
```
Comment (dschneider-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementListResult.java]: make this "final" if this is the only assignment

File: geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java (hunk 1 above)
Comment (jinmeiliao) [reply geode-management/src/main/java/org/apache/geode/management/api/EntityInfo.java]: I think "id" and "groups" seem to go together in this class. I would prefer to make the keys of the json short.

**Pull request merged**